### PR TITLE
docs: create anchors for all header elems & fix a11y in guides

### PR DESCRIPTION
**Pull request by Splaktar**
- Lighthouse's a11y audit wants page headers to progress in order
  - https://web.dev/heading-order/
- Go through all guides and make the headers consistent
  - and ordered to pass the a11y audit
- fix grammar issues and typos

# Before

Screen Shot 2021-01-30 at 03 36 01

### CODE REVIEW

File: guides/theming.md [hunk 1]
```diff
@@ -214,7 +218,8 @@ export class UnicornCandyAppModule {
 }
 ```
 
-#### Theming only certain components
+####Theming only certain components
```
Comment (crisbeto) [inline guides/theming.md:221]: Missing space after the #### here.

File: tools/markdown-to-html/docs-marked-renderer.ts [hunk 2]
```diff
@@ -21,11 +21,11 @@ export class DocsMarkdownRenderer extends Renderer {
 
   /**
    * Transforms a markdown heading into the corresponding HTML output. In our case, we
-   * want to create a header-link for each H3 and H4 heading. This allows users to jump to
+   * want to create a header-link for each H2, H3, and H4 heading. This allows users to jump to
    * specific parts of the docs.
    */
   heading(label: string, level: number, raw: string) {
-    if (level === 3 || level === 4) {
+    if (level === 2 || level === 3 || level === 4) {
```
Comment (crisbeto) [inline tools/markdown-to-html/docs-marked-renderer.ts:28]: Should we add support for h5 and h6 while we're at it?

**Review by crisbeto — APPROVED** [review]
LGTM

File: guides/using-component-harnesses.md [hunk 3]
```diff
@@ -36,7 +36,7 @@ of the CDK's `HarnessEnvironment` and `TestElement` classes.
 
 The foundation for all test harnesses lives in `@angular/cdk/testing`. Start by importing either
 `TestbedHarnessEnvironment` or `ProtractorHarnessEnvironment` based on whether you're writing a
-unit test or an e2e test. From the `HarnessEnvironment`, you can get a `HarnessLoader` instance,
+unit test, or an e2e test. From the `HarnessEnvironment`, you can get a `HarnessLoader` instance,
```
Comment (jelbourn) [inline guides/using-component-harnesses.md:39]: This sentence shouldn't actually have a comma

File: guides/using-component-harnesses.md [hunk 4]
```diff
@@ -235,7 +235,7 @@ self-explanatory.
 
 The harnesses also make clear which option should be selected. Without the harness, you need a
 comment explaining what `options[1]` means, but with the `MatSelectHarness` you can use the `text`
-filter rather than the index and the code becomes self-documenting.
+filter rather than the index, and the code becomes self-documenting.
```
Comment (jelbourn) [inline guides/using-component-harnesses.md:238]: This comma is also misplaced; it was already a bit of a run-on sentence, so I would either leave it as is or rewrite like
```markdown
Without the harness, you need a comment that explains what
`options[1]` means. With `MatSelectHarness`, however, the filter
API makes the code self-documenting.
```

**Review by jelbourn — APPROVED** [review]

**Pull request merged**